SpotifySearch/SpotifySearch/ViewController.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ViewController.swift
+//  SpotifySearch
+//
+//  Created by Kadell on 10/28/16.
+//  Copyright © 2016 Kadell. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    var moreAlbums: Albums?
+    
+    @IBOutlet weak var albumCoverImage: UIImageView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationItem.title = moreAlbums?.nameOfAlbum
```

Comment:
This code is fetching the large image on the main thread. That's why it takes so long to push to the detail view controller as you've noticed. Your code should look like this:

`    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = moreAlbums?.nameOfAlbum
        if let detail = moreAlbums  {
            
            DispatchQueue.global(qos: .background).async {
                guard let imageUrl = URL(string: detail.urlOfImage) else {return }
                let imageData = try! Data(contentsOf: imageUrl)
                
                DispatchQueue.main.async {
                    
                    self.albumCoverImage.image = UIImage(data: imageData)
                    self.albumCoverImage.layer.cornerRadius = 5
                    self.albumCoverImage.layer.masksToBounds = true
                }
            }
        }
    }`

All fetching data on the background, all UI on the main thread.